### ISSUE TITLE
uwsgi with django and pymongo

### ISSUE THREAD

**Issue by Morpho** [unbit/uwsgi#1298]
Sorry, its more of a general question. But we are experiencing many strange problems with uwsi+pymongo. The connection from python to mongodb seems to be droppped every 10 seconds for no reason when there is a bit more web-traffic than usual. Also see my question here: http://stackoverflow.com/questions/37341359/pymongo-with-django-and-uwsgi

Do you have any best-case setup for using django together with mongodb without creating too many connections to mongodb?

**Comment by unbit** [comment]
The only things you should check (being pymongo heavily thread-based) is having this two options:

enable-threads = true
lazy-apps = true

the first one ensure threading is enabled, the second one avoids your threadpools to be opened in the master but not in the workers

**Comment by Morpho** [comment]
Thank you so much for the quick response! That sounds very good. We'll try it out!

**Comment by Morpho** [comment]
We tried it with these settings, unfortunalety it didnt work as expected:
```
[uwsgi]
socket = /var/www/proj/proj.sock
chdir = /var/www/proj
module = proj.wsgi:application
virtualenv = /var/www/virtualenvs/proj/
master = True
lazy-apps = true
buffer-size = 8192
processes = 100
carbon = 127.0.0.1:2003
env = DJANGO_SETTINGS_MODULE=proj.settings
enable-threads = True
single-interpreter = True
uid = www-data
gid = www-data
vacuum = True
disable-logging = true
logger = file:/var/www/proj/log/uwsgi_err.log
```

pymongo opens about 3 connections each process which results in 300 connections. This is too much as mongodb reserves ram for each connection. Can we somehow bundle the connectionpool and have just 10 connections for all 100 processes?

Can you give any hint on how to optimize this setup?

Thanks alot in advance!

**Comment by unbit** [comment]
Unfortunately threadpools do not work this way :(

Probably i would decrese the pool size to 2 and enable one of the cheaper modes to spawn workers only when needed

**Comment by Morpho** [comment]
Thanks for your response! Would you suggest using more threads and less processes maybe? We need a lot of concurrent connections, so we cannot spawn less workers.

**Comment by unbit** [comment]
Yes you could absolutely use more threads and less processes

**Comment by yangmillstheory** [comment]
@unbit can you elaborate on https://github.com/unbit/uwsgi/issues/1298#issuecomment-228061583:

>the second one avoids your threadpools to be opened in the master but not in the workers

**Comment by unbit** [comment]
Yes, the address space of the "parent" process is copied, but the effective threads no more exist in the children
